WebApplicationFactory Assembly EntryPoint validation

caseylobrien opened the pull request:
- Verify generic class used in WebApplicaitonFactory is an entry point to assembly, if not throw InvalidOperationException

Addresses #8250

dnfclas commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/not_signed)](https://cla.dotnetfoundation.org/aspnet/Mvc?pullRequest=8258) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/aspnet/Mvc?pullRequest=8258) before we can accept your contribution.<br/><br/>:x: caseylobrien [sign now](https://cla.dotnetfoundation.org/aspnet/Mvc?pullRequest=8258)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/aspnet/Mvc?pullRequest=8258) it.</sub>

khellang commented on src/Microsoft.AspNetCore.Mvc.Testing/Resources.resx:
```diff
@@ -117,6 +117,9 @@
     <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
   </resheader>
+  <data name="InvalidAssemblyEntryPoint" xml:space="preserve">
+    <value>The provided class '{0}' in Assembly '{1}' is not an entry point to the assembly of the application. A common cause for this error is providing a class from a class library.</value>
```
Type instead of class?

Also, the type itself doesn't need an entry point, it just needs to be in an assembly with an entry point.

caseylobrien replied:
Thoughts on the changes?

javiercn replied:
Type is better

caseylobrien commented:
@khellang Hows that?

pranavkm commented:
@javiercn good to merge?

javiercn commented:
@pranavkm We need to do a small change before we can merge. I'll take care of it this week.

caseylobrien commented:
@javiercn I apologize, I saw that #8250 was closed due to a commit and thought someone else had finished this. Is it possible for me to still make this change? I already moved it but my test was failing because of a missing deps file, and then I got distracted and thought it was closed.

javiercn commented:
@caseylobrien Absolutely! I'll wait for your changes and thanks again for contributing!

javiercn commented on test/WebSites/RazorPagesClassLibrary/xunit.runner.json:
```diff
@@ -0,0 +1,3 @@
+{
```
You shouldn't need this. This is only required on the functional test project. I'll take care of it if you want.

javiercn commented on test/WebSites/RazorPagesClassLibrary/RazorPagesClassLibrary.csproj:
```diff
@@ -12,4 +12,13 @@
+    </None>
+  </ItemGroup>
+  <ItemGroup>
+    <None Include="$(OutDir)\RazorPagesClassLibrary.deps.json" Link="RazorPagesClassLibrary.deps.json" CopyToOutputDirectory="PreserveNewest" />
```
Not sure about this either

javiercn commented:
@caseylobrien Turns out this was a bit trickier than I thought in the beginning. See https://github.com/aspnet/Mvc/commit/ff7c570e7a20c51ede10dfe2838f65951457ed2d.

I can take it from here if you want me too, but I want to give your the opportunity to do the "final push" if you want.

Just let me know if you are ok with me taking over. I'll figure out how to preserve your contribution and thanks a lot again for contributing!

caseylobrien commented:
Thanks @javiercn I appreciate you letting me work on this. The test kept failing due to a missing deps file, which is why I added the extra stuff you removed at an attempt to get that file to copy to the FuncitonalTest projects output directory. Per your comment above it seems like I added 'xunit.runner.json' to the wrong project. I will try that tonight.

javiercn commented:
@caseylobrien This looks great!

caseylobrien commented:
@javiercn Thank you! I am available and interested in assisting in other contributions too. Feel free to message me if other issues pop up.
